### ISSUE TITLE
How do you use DocumentationContextBuilder?

### ISSUE THREAD

**Issue by dplacinta** [springfox/springfox#721]
Hi guys, great work with springfox and the swagger 2.0 implementation, the only problem I see so far is documentation. I'm trying to provide my own ResourceGroupingStrategy, but it doesn't seem to be  picked up from my DocumentationContextBuilder. How do I inject my own ResourceGroupingStrategy and how can I customize my Docket with the DocumentationContextBuilder?

**Comment by dilipkrish** [comment]
Thanks! Yeah we're still trying to improve the documentation. But in the meanwhile, since ResourceGroupingStrategy is a plugin, you could just create your own strategy and give it the right priority (ordering anything less that 0 should work. [example here](https://github.com/springfox/springfox/blob/42036696252d1635d976c463f18e9eadeff05799/springfox-spring-web/src/main/java/springfox/documentation/spring/web/readers/operation/DefaultOperationReader.java#L30) and register it as a bean and it should just pick it up automatically. There is no need to register that with your Docket.

**Comment by dplacinta** [comment]
Hi @dilipkrish,
Thanks for the suggestion, but unfortunately it doesn't work.
I've tried that before posting the question, but if I make my class a @Component, it conflicts with the ClassOrApiAnnotationResourceGrouping bean, when creating bean with name operationTagsReader.

```java
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'operationTagsReader' defined in URL [jar:file:/home/user/.gradle/caches/modules-2/files-2.1/io.springfox/springfox-spring-web/2.0.0/1f35adf85f3a232c94130fc371479a4af89a4e31/springfox-spring-web-2.0.0.jar!/springfox/documentation/spring/web/readers/operation/OperationTagsReader.class]: Unsatisfied dependency expressed through constructor argument with index 0 of type [springfox.documentation.spi.service.ResourceGroupingStrategy]: : No qualifying bean of type [springfox.documentation.spi.service.ResourceGroupingStrategy] is defined: expected single matching bean but found 2: ApiAnnotationOrMappingResourceGroupingStrategy,classOrApiAnnotationResourceGrouping; nested exception is org.springframework.beans.factory.NoUniqueBeanDefinitionException: No qualifying bean of type [springfox.documentation.spi.service.ResourceGroupingStrategy] is defined: expected single matching bean but found 2: ApiAnnotationOrMappingResourceGroupingStrategy,classOrApiAnnotationResourceGrouping
```

**Comment by dilipkrish** [comment]
Yeah that's been fixed in 2.0.1-snapshot. Will be releasing the fix pretty soon. What exactly are u trying to do with a custom bean? There may be a simple solution without needing the implement the strategy yourself.

**Comment by dplacinta** [comment]
It could actually be a bug, but I also want more control over the resource listing (and hopefully the operations listing, although I was unable to achieve that yet).
So, with the default grouping strategy (ClassOrApiAnnotationResourceGrouping), if I have the same value for the @Api("<name>") annotation for 2 different controllers, only the first one gets picked. I'd like to be able to group them by path, pretty much what SpringGroupingStrategy does, but there's an additional feature I require - instead of having the actual values for @Api names and description in the code, I'd like to have keys that I lookup in a properties files, so that the documentation team can work only with the resource files instead of interfering with the code and the resulting swagger 2.0 JSON contains the values in the properties files.
